tests/components/rest/test_switch.py

```diff
@@ -16,254 +16,239 @@
-                        CONF_RESOURCE: "http://localhost",
-                    }
-                },
-            )
-        assert aioclient_mock.call_count == 1
-
-    def test_setup(self, aioclient_mock):
-        """Test setup with valid configuration."""
-        aioclient_mock.get("http://localhost", status=HTTP_OK)
-        assert setup_component(
-            self.hass,
-            SWITCH_DOMAIN,
-            {
-                SWITCH_DOMAIN: {
-                    CONF_PLATFORM: rest.DOMAIN,
-                    CONF_NAME: "foo",
-                    CONF_RESOURCE: "http://localhost",
-                    CONF_HEADERS: {"Content-type": CONTENT_TYPE_JSON},
-                    rest.CONF_BODY_ON: "custom on text",
-                    rest.CONF_BODY_OFF: "custom off text",
-                }
-            },
-        )
-        assert aioclient_mock.call_count == 1
-        assert_setup_component(1, SWITCH_DOMAIN)
-
-    def test_setup_with_state_resource(self, aioclient_mock):
-        """Test setup with valid configuration."""
-        aioclient_mock.get("http://localhost", status=HTTP_NOT_FOUND)
-        aioclient_mock.get("http://localhost/state", status=HTTP_OK)
-        assert setup_component(
-            self.hass,
+from homeassistant.setup import async_setup_component
+
+from tests.common import assert_setup_component
+
+"""Tests for setting up the REST switch platform."""
+
+NAME = "foo"
+METHOD = "post"
+RESOURCE = "http://localhost/"
+STATE_RESOURCE = RESOURCE
+HEADERS = {"Content-type": CONTENT_TYPE_JSON}
+AUTH = None
+
+
+async def test_setup_missing_config(hass):
+    """Test setup with configuration missing required entries."""
+    assert not await rest.async_setup_platform(hass, {CONF_PLATFORM: rest.DOMAIN}, None)
+
+
+async def test_setup_missing_schema(hass):
+    """Test setup with resource missing schema."""
+    assert not await rest.async_setup_platform(
+        hass,
+        {CONF_PLATFORM: rest.DOMAIN, CONF_RESOURCE: "localhost"},
+        None,
+    )
+
+
+async def test_setup_failed_connect(hass, aioclient_mock):
+    """Test setup when connection error occurs."""
+    aioclient_mock.get("http://localhost", exc=aiohttp.ClientError)
+    assert not await rest.async_setup_platform(
+        hass,
+        {CONF_PLATFORM: rest.DOMAIN, CONF_RESOURCE: "http://localhost"},
+        None,
+    )
+
+
+async def test_setup_timeout(hass, aioclient_mock):
+    """Test setup when connection timeout occurs."""
+    aioclient_mock.get("http://localhost", exc=asyncio.TimeoutError())
+    assert not await rest.async_setup_platform(
+        hass,
+        {CONF_PLATFORM: rest.DOMAIN, CONF_RESOURCE: "http://localhost"},
+        None,
+    )
+
+
+async def test_setup_minimum(hass, aioclient_mock):
+    """Test setup with minimum configuration."""
+    aioclient_mock.get("http://localhost", status=HTTP_OK)
+    with assert_setup_component(1, SWITCH_DOMAIN):
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)